significant_strikes.py

```diff
@@ -0,0 +1,500 @@
+            num_fights = len(prev_fights)
+
+            # Initialize empty DataFrames for the scenarios where there are fewer than 3 or 5 fights
+            last_3_fights = pd.DataFrame()
+            last_5_fights = pd.DataFrame()
+
+            # Check if the fighter has at least 3 fights
+            if num_fights >= 3:
+                last_3_fights = prev_fights.tail(3)
+                
+            else:
+                # If fewer than 3 fights, return all available fights
+                last_3_fights = prev_fights
+
+            # Check if the fighter has at least 5 fights
+            if num_fights >= 5:
+                last_5_fights = prev_fights.tail(5)
+                
+            else:
+                # If fewer than 5 fights, return all available fights
+                last_5_fights = prev_fights
+                
+            return last_3_fights, last_5_fights, prev_fights
+
+        return pd.DataFrame(), pd.DataFrame(), pd.DataFrame()
+        
+
+    def get_raw_significant_strikes_stats(self, df, round):
+        """
+        Extracts raw significant strike stats from the dataframe for a specific round.
+
+        Args:
+            df (pd.DataFrame): The dataframe containing fight records.
+            round (str): The round for which the statistics are required.
+
+        Returns:
+            pd.DataFrame: A dataframe with raw stats for the specified round for fighters a and b.
+        """
+
+        # Define column names based on the round
+        if round == "overall":
+            # Use overall stats columns
+            sig_str_cols = [
+                'fighter_a_total_sig_str_landed',
+                'fighter_a_total_sig_str_attempted',
+                'fighter_b_total_sig_str_landed',
+                'fighter_b_total_sig_str_attempted'
+            ]
+        else:
+            # Use specific round stats columns
+            sig_str_cols = [
+                f'fighter_a_round_{round[1]}_sig_str_landed',
+                f'fighter_a_round_{round[1]}_sig_str_attempted',
+                f'fighter_b_round_{round[1]}_sig_str_landed',
+                f'fighter_b_round_{round[1]}_sig_str_attempted'
+            ]
+
+        # Initialize an empty DataFrame with the specified columns
+        stats_df = pd.DataFrame(columns=[
+            'total_fight_time', 'fighter_a_id', 'fighter_b_id',
+            'fighter_a_sig_str_landed', 'fighter_a_sig_str_attempted',
+            'fighter_b_sig_str_landed', 'fighter_b_sig_str_attempted'
+        ])
+
+        # Iterate through each fight in the DataFrame
+        for index, fight in df.iterrows():
+            # Calculate the total fight time
+            total_fight_time = self.get_round_time(fight['outcome_round'], fight['outcome_time'], round)
+
+            # Extract relevant columns for the round
+            sig_str_stats = [fight[col] for col in sig_str_cols]
+
+            # Create a list of stats for the current fight
```

Comment:
Don't forget that the fighter_a in one fight could be fighter_b in the next, so make sure that you're grabbing their stats by a or b and not that fighter_a is getting all the fighter_a stats in the last X fights regardless of whether they are fighter_a in every fight or not

Reply:
yah thats why i get the colomns for both fighters a and b then in the other functions i get the correct fighter's stat based on fighter_id

Reply:
okok fire